com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/util/ApplicationURITest.java

```diff
@@ -24,9 +34,100 @@
+        expectedParts.put(SupportedParameters.ROUTE_PATH, null);
+        Assertions.assertTrue(Maps.difference(expectedParts, applicationURI.getURIParts())
+                                  .areEqual());
+    }
+
+    @Test
+    public void testGetURIPart() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"),
+                                                                            "/custom-path"));
+        assertParametersAreValid("custom-host", applicationURI.getURIPart(SupportedParameters.HOST), "custom-domain",
+                                 applicationURI.getURIPart(SupportedParameters.DOMAIN), "/custom-path",
+                                 applicationURI.getURIPart(SupportedParameters.ROUTE_PATH));
+        Assertions.assertNull(applicationURI.getURIPart("invalid-parameter"));
+    }
+
+    @Test
+    public void testURIPart() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"),
+                                                                            "custom-path"));
+        applicationURI.setURIPart(SupportedParameters.HOST, "custom-host-1");
+        applicationURI.setURIPart(SupportedParameters.DOMAIN, "custom-domain-1");
+        applicationURI.setURIPart(SupportedParameters.ROUTE_PATH, "/custom-path-1");
+        applicationURI.setURIPart("invalid-parameter", "value");
+        assertParametersAreValid("custom-host-1", applicationURI.getURIPart(SupportedParameters.HOST), "custom-domain-1",
+                                 applicationURI.getURIPart(SupportedParameters.DOMAIN), "/custom-path-1",
+                                 applicationURI.getURIPart(SupportedParameters.ROUTE_PATH));
+    }
+
+    @Test
+    public void testToStringWithValidHostAndPath() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"),
+                                                                            "/custom-path"));
+        Assertions.assertEquals("custom-host.custom-domain/custom-path", applicationURI.toString());
+    }
+
+    @Test
+    public void testToStringWithValidHostAndWithoutPath() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("custom-host", createCloudDomain("custom-domain"), null));
+        Assertions.assertEquals("custom-host.custom-domain", applicationURI.toString());
+    }
+
+    @Test
+    public void testToStringWithoutHostAndWithoutPath() {
+        ApplicationURI applicationURI = new ApplicationURI(createCloudRoute("", createCloudDomain("custom-domain"), null));
+        Assertions.assertEquals("custom-domain", applicationURI.toString());
+    }
+
+    private void assertParametersAreValid(String expectedHost, Object host, String expectedDomain, Object domain, String expectedPath,
```

Comment:
Split into multiple methods

Reply:
I prefer to use Assertions.assertEquels instead of splitting to three methods which will do the same.